### ISSUE TITLE
Net 4.6 compatibility

### ISSUE THREAD

**Issue by rgjones33** [migueldeicaza/TensorFlowSharp#258]
@migueldeicaza Is there a hard dependency anywhere in the code for NET 4.6.1 which is required by SystemValue tuple 4.4.0? Could we back level System.ValueTuple to 4.3.1 without breaking anything? Trying to make this package compatible with Unity which uses Net.Framework 4.6

"Warning The primary reference "TensorFlowSharp" could not be resolved because it was built against the ".NETFramework,Version=v4.6.1" framework. This is a higher version than the currently targeted framework ".NETFramework,Version=v4.6"."

**Comment by migueldeicaza** [comment]
I dont think so, you could try to lower the version number and submit a Pull Request, and that will validate the change.

**Comment by edwardyehuang** [comment]
Well, I did a simple approach to make it compatible with.NET.4.5 (Visual Studio 2013) for one of my customers. I recompiled the TensorflowSharp in 4.5 and replace the new 4.6.1 API with the code copied from .NET 4.6.1. However, it will be slower than the official .NET 4.6, technically speaking, but it worked.

**Comment by bendiy** [comment]
@edwardyehuang 
Do you have this TensorflowSharp source in .NET 4.5 version anywhere? I checked your fork, but didn't see anything.

Thanks!

**Comment by narend** [comment]
I also need tensorflowsharp in 4.5 for a customer. @bendiy  @edwardyehuang  Do any of you have 4.5 version? I checked your forks, but did not find anything. If you don't have, I can take a shot at doing it.

**Comment by narend** [comment]
@migueldeicaza , most changes seem to be replacing tuples with Record<> class to make it build in 4.5. Shall I try this in my fork and would you be willing to take this change? We can look at conditional compilation as well.
